org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -1234,5 +1235,45 @@ public void checkOppositeReferenceUsed(Assignment assignment) {
 			}
 		}
 	}
-
+	
+	@Check
+	public void checkAnnotations(AbstractRule rule) {
```

Comment:
We should remove this check, because the reason I asked for generic annotations is to allow users to add arbitrary ones without changing our code. E.g. to use them in custom generator fragments.